tools/mot_error_visualization.py

```diff
@@ -0,0 +1,265 @@
+                      thickness=2,
+                      font_scale=0.4,
+                      text_width=10,
+                      text_height=15,
+                      show=False,
+                      wait_time=0,
+                      out_file=None):
+    """Show the wrong tracks with opencv.
+
+    Args:
+        img (str or ndarray): The image to be displayed.
+        bboxes (ndarray): A ndarray of shape (k, 5).
+        ids (ndarray): A ndarray of shape (k, ).
+        wrong_types (ndarray): A ndarray of shape (k, ), where 0 denotes
+            false positives, 1 denotes false negative and 2 denotes ID switch.
+        bbox_colors (list[tuple], optional): A list of colors to
+            draw boxes with different wrong type. Defaults to None.
+        thickness (int, optional): Thickness of lines.
+            Defaults to 2.
+        font_scale (float, optional): Font scale to draw id and score.
+            Defaults to 0.4.
+        text_width (int, optional): Width to draw id and score.
+            Defaults to 10.
+        text_height (int, optional): Height to draw id and score.
+            Defaults to 15.
+        show (bool, optional): Whether to show the image on the fly.
+            Defaults to False.
+        wait_time (int, optional): Value of waitKey param.
+            Defaults to 0.
+        out_file (str, optional): The filename to write the image.
+            Defaults to None.
+    """
+    assert bboxes.ndim == 2
+    assert ids.ndim == 1
+    assert wrong_types.ndim == 1
+    assert bboxes.shape[1] == 5
+
+    if bbox_colors:
+        assert len(bbox_colors) == 3
+    else:
+        bbox_colors = sns.color_palette(n_colors=3)
+        bbox_colors = [[int(255 * _c) for _c in bbox_color][::-1]
+                       for bbox_color in bbox_colors]
+    if isinstance(img, str):
+        img = mmcv.imread(img)
+
+    img_shape = img.shape
+    bboxes[:, 0::2] = np.clip(bboxes[:, 0::2], 0, img_shape[1])
+    bboxes[:, 1::2] = np.clip(bboxes[:, 1::2], 0, img_shape[0])
+
+    for bbox, wrong_type, id in zip(bboxes, wrong_types, ids):
+        x1, y1, x2, y2 = bbox[:4].astype(np.int32)
+        score = float(bbox[-1])
+
+        # bbox
+        bbox_color = bbox_colors[wrong_type]
+        cv2.rectangle(img, (x1, y1), (x2, y2), bbox_color, thickness=thickness)
+
+        # FN does not have id and score
+        if wrong_type == 1:
+            continue
+
+        # id
+        text = str(id)
+        width = len(text) * text_width
+        img[y1:y1 + text_height, x1:x1 + width, :] = bbox_color
+        cv2.putText(
+            img,
+            str(id), (x1, y1 + text_height - 2),
+            cv2.FONT_HERSHEY_COMPLEX,
+            font_scale,
+            color=(0, 0, 0))
+
+        # score
+        text = '{:.02f}'.format(score)
+        width = len(text) * text_width
+        img[y1 - text_height:y1, x1:x1 + width, :] = bbox_color
+        cv2.putText(
+            img,
+            text, (x1, y1 - 2),
```

Comment:
If the top left corner of bounding box is on the border, we will not see the score.
We can put both id and score inside the box. The text need add the prefix of `id:` or `score:`  to avoid ambiguity.